src/OrleansLoggerProvider/OrleansLoggingFactoryExtension.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.Extensions.Logging;
+using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Net;
+
+namespace Orleans.Extensions.Logging
+{
+    public static class OrleansLoggingFactoryExtension
+    {
+        /// <summary>
+        /// Add message bulking feature onto <param name="provider"></param>, and add that new logger provider 
+        /// <see cref="MessageBulkingLoggerProvider{TDecoratedLoggerProvider}"/> into <param name="builder"></param>.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="provider"></param>
+        /// <param name="messageBulkingConfig"></param>
+        /// <returns></returns>
+        public static ILoggingBuilder AddMesageBulkingLoggerProvider<TDecoratedLoggerProvider>(this ILoggingBuilder builder, TDecoratedLoggerProvider provider, MessageBulkingConfig messageBulkingConfig)
+            where TDecoratedLoggerProvider : ILoggerProvider
+        {
+            builder.AddProvider(new MessageBulkingLoggerProvider<TDecoratedLoggerProvider>(provider, messageBulkingConfig));
+            return builder;
+        }
+
+        /// <summary>
+        /// Add <see cref="OrleansLoggerProvider"> with message bulking feature to LoggerFactory
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="logConsumers">log consumers which user want to write log events to</param>
+        /// <param name="messageBulkingConfig">config for message bulking feature</param>
+        /// <returns></returns>
+        [Obsolete]
+        public static ILoggingBuilder AddOrleansLogging(
+            this ILoggingBuilder builder,
+            List<ILogConsumer> logConsumers,
+            IPEndPoint ipEndPoint = null,
```

Comment:
hmmm, this ipEndPoint here might be tricky. We can probably address in a different PR, but maybe a solution is to add extension methods for ISiloBuilder and IClientBuilder also, so that we can resolve the IP endpoint first and then add the provider, but not sure about order of initialization.

Reply:
we can discuss this in another PR. But IMO, this extensions should be decoupled with core interface as much as possible, since it is being deprecated. I see we provide convenience for users on how to resolve IpEndpoint using extenion method on ISiloBuilder, but at the mean time, we also couple those two closer than before, event it is just an extension method. I'm not concerned about resolving ipEndpoint, since it is pretty easy for user to resolve it as in they are the ones who configure that,.

Reply:
yes, let's discuss afterwards. Although it might not be as easy to know the ipendpoint though, as it might be resolved and bound by the silo itself, not the user